libs/entitlement/src/main/java/org/elasticsearch/entitlement/runtime/policy/PolicyIllegalArgumentException.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+package org.elasticsearch.entitlement.runtime.policy;
+
+public class PolicyIllegalArgumentException extends IllegalArgumentException {
```

Comment:
Could this extend RuntimeException? There doesn't seem anything inherent or critical about IAE here. I also wonder if we could also call it something like `PolicyValidationException`?

Reply:
Thought it might have been useful if we caught `IllegalArgumentException` elsewhere, but I will make this change.

Reply:
Thought it might have been useful if we caught `IllegalArgumentException` elsewhere, but I will make this change.